### PR TITLE
Fix build system to allow CMAKE_<COMPILER>_FLAGS to be specified for submodules

### LINKED ISSUE #2050: Fix build system so that CMAKE_<COMPILER>_FLAGS set at top level will be used by subcomponents
## Description

CMEPS, CDEPS,  CICE, and MOM6 do not inherit CMAKE_<COMPILER>_FLAGS from the top level build and there is no way to specify new flags like "-fPIC" when linking to other applications like JEDI>

## Solution
Change CMakeLists.txt files to inherit flags from top level

## Alternatives
The "-fPIC" flag could be added to each CMakeLists.txt file directly, but that would not allow for any flexibility

### PR DESCRIPTION
**Pull request by mark-a-potts**
## PR Author Checklist:

- [X] I have linked PR's from all sub-components involved in section below. 
- [X] I have run the full RT suite on either Hera/Hercules AND have attached the log to this PR below this line:
  - LOG:  committed to branch
- [ ] I have added the list of all failed regression tests to "Anticipated changes" section.
- [X] I have filled out all sections of the template.

## Description
<--Provide a detailed description of what this PR does in the space provided below-->
This PR allows the CMAKE_\<COMPILER\>_FLAGS specified on the build line for the UFS-WM to be passed down to CMEPS, CDEPS, CICE and MOM6 components. This allows flags like "-fPIC" to be specified and used when building the UFS as a library to be used with JEDI applications.

This PR changes lines in associated CMakeLists.txt files from 

set(CMAKE_\<COMPILER\>_FLAGS "-g -some-other-flag") 

to 

set(CMAKE_\<COMPILER\>_FLAGS "${CMAKE_\<COMPILER\>_FLAGS} -g -some-other-flag") 

which preserves the flags which are set at the top level and appends to them instead of overwriting the top level flags.

## Commit Message

Allow CMAKE_\<COMPILER\>_FLAGS to be passed to sub-components for use with JEDI.

### Associated UFSWM Issue to close
- Closes #2050

### Subcomponent Pull Requests
None

### Blocking Dependencies
None

### Subcomponents involved:
- [ ] AQM
- [X] CDEPS
- [X] CICE
- [X] CMEPS
- [ ] CMakeModules
- [ ] FV3
- [ ] GOCART
- [ ] HYCOM
- [X] MOM6
- [ ] NOAHMP
- [ ] WW3
- [ ] stochastic_physics
- [ ] none

### Input data
- [X] No changes are expected to input data.
- [ ] Changes are expected to input data:
  - [ ] New input data.
  - [ ] Updated input data.

### Regression Tests:
- [X] No changes are expected to any regression test.
- [ ] Changes are expected to the following tests:

<details><summary>Tests effected by changes in this PR:</summary>

</details>

### Libraries

- [X] Not Needed
- [ ] Needed
  - [ ] Create separate issue in [JCSDA/spack-stack](https://github.com/JCSDA/spack-stack) asking for update to library. Include library name, library version.
  - [ ] Add issue link from JCSDA/spack-stack following this item 

<details><summary>Code Managers Log</summary>

- [ ] This PR is up-to-date with the top of all sub-component repositories except for those sub-components which are the subject of this PR.
- [ ] Move new/updated input data on RDHPCS Hera and propagate input data changes to all supported systems.
  - [ ] N/A

### Testing Log:
- RDHPCS
  - [ ] Hera
  - [ ] Orion
  - [ ] Hercules
  - [ ] Jet
  - [ ] Gaea
  - [ ] Derecho
- WCOSS2
  - [ ] Dogwood/Cactus
  - [ ] Acorn
- CI
  - [ ] Completed
- opnReqTest
  - [ ] N/A
  - [ ] Log attached to comment
</details>

### CODE REVIEW

**Comment by DeniseWorthen** [comment]
@mark-a-potts We're adding a commit message to the PR templates and I'm manually adding it to current/pending PRs. Please add in the space provided. Thanks.

**Comment by mark-a-potts** [comment]
Hi @DeniseWorthen  Can you tell me what the commit message should say and where to add it? I am seeing the space provided. Thanks!

**Comment by DeniseWorthen** [comment]
@mark-a-potts You need to "edit" the PR description. Look in the upper right corner of your PR (under the 'files changed') and use the ... to edit the page. Scroll down until you see the html tag about entering your Commit Message. Or, you can write it here and I will do this for you.

Generally, the commit message should be a succinct description of what the PR does. For example, you might say something like "add capabliltity to pass CMAKE flags down to components for use by JEDI".

This will be used by code managers when they squash merge to give users a quick understanding of what the commit changed. We'd like developers to write the commit message, since they are the ones who know what is important.

**Comment by mark-a-potts** [comment]
Thanks @DeniseWorthen! Done.

File: CICE-interface/CMakeLists.txt [hunk 1]
```diff
@@ -20,20 +20,17 @@ endif()
 
 ### CICE C compiler flags
 if(CMAKE_C_COMPILER_ID MATCHES "GNU")
-  set(CMAKE_C_FLAGS "-g")
-  set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS}")
+  set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -g")
```
Comment (junwang-noaa) [inline CICE-interface/CMakeLists.txt:23]: Not sure if this change is required, or is it just for consistency?

**Review by junwang-noaa — DISMISSED** [review]
I assume we only need to compile these components as shared library. E.g.  GOCART is not included at this time.

**Comment by mark-a-potts** [comment]
I just checked, and it looks like GOCART does not need a change like this, but AQM and HYCOM have similar issues. WW3 specifically sets the CMAKE_Fortran_FLAGS to "", so that may require a different fix if/when JEDI needs to link to UFS with WW3 enabled. Should I update AQM and HYCOM to match these changes?

**Comment by junwang-noaa** [comment]
I think we can do it later when these components need to be compiled as libraries within JEDI.

**Comment by jkbk2004** [comment]
merged with #2013

**Pull request closed without merging**